Add PropertyTerm DTOs and their unit tests

marierinden opened the pull request:
---

### Description

This pull request introduces the `PropertyTermRequest`, `PropertyTermResponse`, and `PropertyTermCollectionResponse` DTOs for better structured handling of property terms. It includes integration with `PropertyTermResource` to enable CRUD operations and property term retrieval.

Additionally, unit tests were added to validate the behavior of these DTOs, ensuring accurate data structures, types, and mappings.

marierinden committed:
Add PropertyTerm DTOs and Resource integration

Introduce PropertyTermRequest, PropertyTermResponse, and PropertyTermCollectionResponse DTOs along with PropertyTermResource. These additions enable CRUD operations and property term retrieval through structured requests and responses.

marierinden committed:
Add unit tests for PropertyTerm DTO requests and responses

Introduce unit tests for PropertyTermRequest, PropertyTermResponse, and PropertyTermCollectionResponse to ensure correctness of their behavior. These tests validate data structures, expected types, and data mappings for the PropertyTerm DTOs.

fredrik-tveraaen-apiera commented on src/DTO/Request/PropertyTerm/PropertyTermRequest.php:
```diff
@@ -0,0 +1,60 @@
+        #[RequestField('property')]
+        private string $property,
+        #[RequestField('store')]
+        private string $store,
```
Remove store as the resource is nested under a store nested resource

fredrik-tveraaen-apiera commented on src/Resource/PropertyTermResource.php:
```diff
@@ -0,0 +1,175 @@
+ */
+final readonly class PropertyTermResource implements RequestResourceInterface
+{
+    private const string ENDPOINT = '/propertyTerms';
```
Should be /terms

marierinden committed:
Refactor PropertyTermRequest and update PropertyTermResource endpoint.

Remove the `store` field from PropertyTermRequest and update related methods. Adjust PropertyTermResource to use `property` instead and change the endpoint to `/terms`. Update corresponding tests and serialization handling.

marierinden committed:
Remove `store` field and usage from PropertyTerm entities.

The `store` field was removed from `PropertyTermResponse` and related tests, as it is no longer needed. Updated exception messages to correctly reference `Property IRI` instead of `Store IRI` for improved clarity.

fredrik-tveraaen-apiera committed:
Add support for PropertyTerm resource and adjust related logic

Updated PropertyTerm DTOs to accept nullable fields and adjusted their behavior in `toArray`. Added PropertyTerm handling in `LdType` and updated test cases accordingly. Introduced `propertyTerm` method in `ApieraSdk` to enable PropertyTerm resource usage.

fredrik-tveraaen-apiera committed:
Refactor PropertyTermResponse to use ResponseField exclusively

Removed RequestField attributes in favor of ResponseField for "name" and "property." Added a new "store" property with corresponding ResponseField and accessor method. Updated related test cases to reflect the changes.

fredrik-tveraaen-apiera committed:
Add integration tests for PropertyTerm CRUD operations

This commit introduces tests for create, read, update, and delete operations for the PropertyTerm resource. Each operation has a dedicated test class implementing resource-specific logic and validation to ensure proper behavior. These tests ensure all PropertyTerm CRUD functionalities are correctly integrated and scoped.

fredrik-tveraaen-apiera committed:
Merge origin release/0.3.0

fredrik-tveraaen-apiera reviewed (approved):
:100:
